### ISSUE TITLE
Improvements to layout

### ISSUE THREAD

**Issue by aliabid94** [gradio-app/gradio#4010]
Some small things that could improve layouts a lot IMO.

1. Every component should have `scale` and `min-width` args to control their layout within a Row. 
Right now, to control these, you have to nest them within gr.Columns, such as 

```python
with gr.Row():
  with gr.Column(scale=1):
    text1 = gr.Textbox()
  with gr.Column(scale=2):
    text2 = gr.Textbox()
```

This breaks how adjacent field components naturally connect, and also breaks the alignment of heights of adjacent elements in a Row. Instead,

```python
with gr.Row():
    text1 = gr.Textbox(scale=1)
    text2 = gr.Textbox(scale=2)
```

fixes the above problems, and is cleaner to write. We can also kill the `full_width` arg, which currently exists only for the Button, and replace that with `scale=0` or `scale=1`.

2. Move `.style()` args to the main constructor.

There's no real logic what goes in style and what goes in the constructor. For example, `show_label` we put in the constructor, but `show_container` goes in style. We also don't provide access to the style args in `.update` but we do provide access to the constructor args.  Cleaner to have it all in the constructor

3. Figure out the look of adjacent elements in `gr.Box()` or with `gr.Textbox().style(show_container=False)`.

I don't know why people keep using these. Adjacent elements have weird rounding in both scenarios. An example from deep floyd space that uses both:

<img width="579" alt="Screen Shot 2023-04-28 at 12 35 45 PM" src="https://user-images.githubusercontent.com/7870876/235238010-37fbeaf3-9372-41f4-a7b6-29ee6d10d7de.png">

**Comment by freddyaboulton** [comment]
Yes agreed @aliabid94 ! I think this would close https://github.com/gradio-app/gradio/issues/3202 as well

**Comment by abidlabs** [comment]
Agreed with all three points. (2) in included this issue as well: https://github.com/gradio-app/gradio/issues/3438, and should be part of 4.0. Should we just add this issue to the 4.0 milestone as well?

**Comment by abidlabs** [comment]
Looks like all three are closed now, though #3202 is not closed. Thanks @aliabid94!
